### PR TITLE
doc: extend javadoc for JavadocTokenTypes.RULE_TYPES_OFFSET

### PR DESCRIPTION
**Pull request by romani**
extension of doc, in scope of #5114.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 1]
```diff
@@ -1230,13 +1230,22 @@
...
 
-    /** Rule types offset. */
-    private static final int RULE_TYPES_OFFSET = 10000;
-
     //--------------------------------------------------------------------------------------------//
     //------- JAVADOC TAGS DEPENDING ON RULE TYPES OFFSET ----------------------------------------//
     //--------------------------------------------------------------------------------------------//
 
+    /** Rule types offset.
+     * RULE_TYPES_OFFSET constant is used to split lexer tokens types и parser rules types.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: > и 

Should this be `and`?

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 2]
```diff
@@ -1230,13 +1230,22 @@
...
+     * We need unique numbers for all tokens,
+     * ANTLR do not need this and that is why his types are mixed by used values.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: > why his types 

should `his` be `this`?

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 3]
```diff
@@ -1230,13 +1230,22 @@
...
+     * All values we can take a look at
+     * target/generated-sources/antlr/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParser.java
+     * For example: LEADING_ASTERISK=1 and RULE_htmlElement = 1.
+     * RULE_TYPES_OFFSET required to shift parser rules,
+     * to let them not overlap with types that have prefix "RULE_".
+     * */
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: change to just `*/`.

**Comment by romani** [comment]
@rnveach , thanks a lot , fixed, rebased, merged manually

**Pull request closed without merging**